File: 6-solution-kotlin-mixed/src/main/java/de/berenberg/schulung/repository/RoomRepository.java
package de.berenberg.schulung.repository;

import de.berenberg.schulung.domain.Building;
import de.berenberg.schulung.domain.Room;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RoomRepository extends JpaRepository<Room, Long> {

    List<Room> findByBuilding(Building building);

}

File: 6-solution-kotlin-only/src/main/kotlin/de/berenberg/schulung/repository/RoomRepository.kt
package de.berenberg.schulung.repository

import de.berenberg.schulung.domain.Building
import de.berenberg.schulung.domain.Room
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

interface RoomRepository : JpaRepository<Room, Long> {
    fun findByBuilding(building: Building): List<Room>
}
